File: src/main/java/com/hedvig/backoffice/services/account/dto/ApproveChargeRequestDto.java
package com.hedvig.backoffice.services.account.dto;

import com.hedvig.backoffice.graphql.types.MemberChargeApproval;
import lombok.Value;

import javax.money.MonetaryAmount;
import java.util.List;

@Value
public class ApproveChargeRequestDto {
  String memberId;
  MonetaryAmount amount;

  public static ApproveChargeRequestDto from(MemberChargeApproval memberChargeApproval) {
    return new ApproveChargeRequestDto(
      memberChargeApproval.getMemberId(),
      memberChargeApproval.getAmount()
    );
  }
}

File: src/main/java/com/hedvig/backoffice/services/account/dto/ApproveChargeRequestDto.kt
package com.hedvig.backoffice.services.account.dto

import com.hedvig.backoffice.graphql.types.MemberChargeApproval
import lombok.Value
import javax.money.MonetaryAmount

@Value
class ApproveChargeRequestDto(
    val memberId: String,
    val amount: MonetaryAmount
) {
    companion object {
        fun from(memberChargeApproval: MemberChargeApproval): ApproveChargeRequestDto {
            return ApproveChargeRequestDto(
                memberChargeApproval.memberId,
                memberChargeApproval.amount
            )
        }
    }
}
